pkg/distsql/select_result.go

```diff
@@ -603,11 +603,13 @@ func (r *selectResult) Close() error {
 	if respSize > 0 {
 		r.memConsume(-respSize)
 	}
-	if unconsumed, ok := r.resp.(copr.HasUnconsumedCopRuntimeStats); ok && unconsumed != nil {
-		unconsumedCopStats := unconsumed.CollectUnconsumedCopRuntimeStats()
-		for _, copStats := range unconsumedCopStats {
-			_ = r.updateCopRuntimeStats(context.Background(), copStats, time.Duration(0))
-			r.ctx.GetSessionVars().StmtCtx.MergeExecDetails(&copStats.ExecDetails, nil)
+	if r.ctx != nil {
```

Comment:
If `r.ctx == nil`, L624 also panics.

Reply:
May there is some relationship between `r.stats != nil` and `r.ctx != nil`, because L624 is not new introduced and not sure if it ever paniced before.

Reply:
Add  judgement `r.ctx != nil` to check it by https://github.com/pingcap/tidb/pull/51950/commits/8e0d4ccd7c2ae0022907d0b74763e89c3472462c.

Reply:
I see, both ctx and stats are not set for check sum requests.